scripts/deploy/tasks.py

```diff
@@ -6,166 +6,267 @@
-            command = ['sudo', 'bash', '-c', f'echo "127.0.0.1 {host}" >> /etc/hosts']
+            command = ["sudo", "bash", "-c", f'echo "127.0.0.1 {host}" >> /etc/hosts']
             subprocess.run(command)
     else:
-        print("Cluster type is not k3s-k3d or ingress choice is not my-ip-co. Skipping /etc/hosts setup.")
+        print(
+            "Cluster type is not k3s-k3d or ingress choice is not my-ip-co. Skipping /etc/hosts setup."
+        )
+
 
 @task
 def add_route53_entry(c, f):  # NOSONAR
     values = load_values(c, f)
-    host = values.get('ingress', {}).get('host', '')
-    load_balancer = values.get('ingress', {}).get('load_balancer', '')
-    hosted_zone_id = values.get('ingress', {}).get('hostedZoneId', '')
+    host = values.get("ingress", {}).get("host", "")
+    load_balancer = values.get("ingress", {}).get("load_balancer", "")
+    hosted_zone_id = values.get("ingress", {}).get("hostedZoneId", "")
 
     if host and load_balancer and hosted_zone_id:
         # Check if the record already exists
-        check_cmd = f"aws route53 list-resource-record-sets --hosted-zone-id {hosted_zone_id}"
+        check_cmd = (
+            f"aws route53 list-resource-record-sets --hosted-zone-id {hosted_zone_id}"
+        )
         result = subprocess.run(check_cmd, shell=True, capture_output=True, text=True)
         records = json.loads(result.stdout)["ResourceRecordSets"]
 
-        record_exists = any(record["Name"] == host and record["Type"] == "CNAME" for record in records)
+        record_exists = any(
+            record["Name"] == host and record["Type"] == "CNAME" for record in records
+        )
         if not record_exists:
             # Add the record
-            add_cmd = f"aws route53 change-resource-record-sets --hosted-zone-id {hosted_zone_id} --change-batch '{{\"Changes\": [{{\"Action\": \"CREATE\", \"ResourceRecordSet\": {{\"Name\": \"{host}\", \"Type\": \"CNAME\", \"TTL\": 300, \"ResourceRecords\": [{{\"Value\": \"{load_balancer}\"}}]}}}}]}}'"
+            add_cmd = f'aws route53 change-resource-record-sets --hosted-zone-id {hosted_zone_id} --change-batch \'{{"Changes": [{{"Action": "CREATE", "ResourceRecordSet": {{"Name": "{host}", "Type": "CNAME", "TTL": 300, "ResourceRecords": [{{"Value": "{load_balancer}"}}]}}}}]}}\''
             subprocess.run(add_cmd, shell=True)
             print(f"Route53 entry added for {host}")
         else:
             print(f"Route53 entry for {host} already exists")
 
+
 @task
-def helm_install_or_upgrade(c, f, namespace, values, image_tag): # NOSONAR
+def helm_install_or_upgrade(c, f, namespace, values, image_tag):  # NOSONAR
     script_dir = os.path.dirname(os.path.abspath(__file__))
-    chart_filename = "cht-chart-4.x.tgz"
-    project_name = values.get('project_name', '')
-    release_exists = subprocess.run(["helm", "list", "-n", namespace], capture_output=True, text=True).stdout
+    chart_filename = "cht-chart"
```

Comment:
Thanks for the clarification. It wasn't clear that the `tgz` helm chart would be generated from a Github action. I was worried that the raw templates would otherwise have fallen out of sync with the compressed `tgz` file here. I assume you are referring to having the npm package pull the helm chart from https://github.com/medic/helm-charts once the chart is uploaded and the repo is configured as a helm repo?